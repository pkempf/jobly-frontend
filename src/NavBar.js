import React, { useContext } from "react";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import { Link } from "react-router-dom";
import UserContext from "./UserContext";

const NavBar = () => {
  const user = useContext(UserContext);

  return (
    <Navbar bg="primary" variant="dark" expand="md">
      <Navbar.Brand as={Link} to="/">
        Jobly
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="jobly-navbar-nav" />
      <Navbar.Collapse className="justify-content-end" id="jobly-navbar-nav">
        {user.username ? (
          <Nav>
            <Nav.Link as={Link} to="/companies">
              Companies
            </Nav.Link>
            <Nav.Link as={Link} to="/jobs">
              Jobs
            </Nav.Link>
            <Nav.Link as={Link} to="/profile">
              Profile
            </Nav.Link>
            <Nav.Link as={Link} to="/logout">
              Log out {user.username}
            </Nav.Link>
          </Nav>
        ) : (
          <Nav>
            <Nav.Link as={Link} to="/login">
              Login
            </Nav.Link>
            <Nav.Link as={Link} to="/signup">
              Sign Up
            </Nav.Link>
          </Nav>
        )}
      </Navbar.Collapse>
    </Navbar>
  );
};

export default NavBar;
