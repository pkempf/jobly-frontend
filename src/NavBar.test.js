import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar";
import UserContext from "./UserContext";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: "test" }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: "test" }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
