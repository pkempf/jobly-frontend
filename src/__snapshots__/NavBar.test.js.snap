// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-dark bg-primary"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Jobly
    </a>
    <button
      aria-controls="jobly-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="justify-content-end navbar-collapse collapse"
      id="jobly-navbar-nav"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-link"
          href="/companies"
        >
          Companies
        </a>
        <a
          class="nav-link"
          href="/jobs"
        >
          Jobs
        </a>
        <a
          class="nav-link"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="nav-link"
          href="/logout"
        >
          Log out test
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
